deepchem/models/layers.py

```diff
@@ -429,9 +427,9 @@ def _cosine_dist(x, y):
   y: tf.Tensor
     Input Tensor
```

Comment:
Can you add details to the docstring with the types of the tensors being processed and details about the tensor being returned and add a usage example to the docstring?